drivers/serial.c

```diff
@@ -37,14 +37,45 @@ static struct {
     unsigned init;
 } input_state;
 
-extern io_port_t com_ports[2];
+io_port_t __data_rmode com_ports[4];
+
+static const char *com_port_names[] = {
+    [COM1_PORT] = "COM1",
+    [COM2_PORT] = "COM2",
+    [COM3_PORT] = "COM3",
+    [COM4_PORT] = "COM4",
+};
+
+static const uint8_t com_frame_size_values[] = {
+    [COM_FRAME_SIZE_5_BITS] = 5,
+    [COM_FRAME_SIZE_6_BITS] = 6,
+    [COM_FRAME_SIZE_7_BITS] = 7,
+    [COM_FRAME_SIZE_8_BITS] = 8,
+};
+
+static const char com_parity_names[] = {
+    [COM_NO_PARITY] = 'n',   [COM_ODD_PARITY] = 'o', [COM_EVEN_PARITY] = 'e',
+    [COM_HIGH_PARITY] = 'h', [COM_LOW_PARITY] = 'l',
+};
+
+#define COM_STOP_BIT_VALUE(cfg) ((cfg)->stop_bit + 1)
+
+void display_uart_config(const uart_config_t *cfg) {
+    printk("[%s] 0x%x %u,%u%c%u\n", com_port_names[cfg->port], cfg->port, cfg->baud,
```

Comment:
`com_port_names[cfg->port]` needs to be changed accordingly. Maybe simply introduce a helper that maps a COM's port to its number?

Reply:
Fixed by adding a helper function.